src/FakeItEasy/Core/FakeManager.ObjectMemberRule.cs

```diff
@@ -70,7 +80,8 @@ private static bool IsObjetMethod(IFakeObjectCall fakeObjectCall)
 
             private bool TryHandleGetHashCode(IInterceptedFakeObjectCall fakeObjectCall)
             {
-                if (!IsSameMethod(fakeObjectCall.Method, ObjectMethods[2]))
+                if (!IsSameMethod(fakeObjectCall.Method, ObjectGetHashCodeMethod) &&
+                    !IsSameMethod(fakeObjectCall.Method, FakeObjectGetHashCodeMethod))

```

Comment:
I thought so too... indexes were manageable with only 3 methods, but not so much with 6 methods.